metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -414,12 +453,23 @@ def emit_mcp(
 
             payload = json.dumps(payload_dict)
 
-            self._emit_generic(url, payload)
+            response = self._emit_generic(url, payload)
+
+            if self._should_trace(async_flag, trace_flag):
+                trace_data = extract_trace_data_from_mcps(response, [mcp])
```

Comment:
any chance of `response` being `None` here?

as few lines above:

```suggestion
                trace_data = extract_trace_data_from_mcps(response, [mcp]) if response else None
```